client/testmessage.go

```diff
@@ -6,18 +6,27 @@ import (
 	"strings"
 
 	"github.com/mitch000001/go-hbci/charset"
+	"golang.org/x/text/encoding/charmap"
 )
 
+func mustEncode(str string, err error) []byte {
+	if err != nil {
+		panic(err)
+	}
+	return []byte(str)
+}
+
 func encryptedTestMessage(dialogID string, encryptedData ...string) []byte {
+	iso8859Encoder := charmap.ISO8859_1.NewEncoder()
 	encryptionHeader := "HNVSK:998:2:+998+1+1::0+1:20150713:173634+2:2:13:@8@\x00\x00\x00\x00\x00\x00\x00\x00:5:1:+280:10000000:12345:V:0:0+0+'"
-	encryptionData := fmt.Sprintf("HNVSD:999:1:+@%d@%s'", len(charset.ToISO8859_1(strings.Join(encryptedData, ""))), charset.ToISO8859_1(strings.Join(encryptedData, "")))
+	encryptionData := fmt.Sprintf("HNVSD:999:1:+@%d@%s'", len(charset.ToISO8859_1(strings.Join(encryptedData, ""))), mustEncode(iso8859Encoder.String(strings.Join(encryptedData, ""))))
```

Comment:
I think pulling this line apart with multiple temporary variables would increase readability -> FTW